wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -1202,6 +1204,41 @@ instance BuildableSafeGen AddressValidity where
     buildSafeGen _ AddressValidity{..} =
         bprint ("{ valid="%build%" }") isValid
 
+-- | An address is either recognised as "ours" or not. An address that is not
+--   recognised may still be ours e.g. an address generated by another wallet instance
+--   will not be considered "ours" until the relevant transaction is confirmed.
+--
+--   In other words, `AddressNotRecognised` makes an inconclusive statement about
+--   an address, whereas `AddressIsOurs` is unambiguous.
+data AddressIsOurs = AddressIsOurs
+                   | AddressNotRecognised
```

Comment:
what about reflecting the comment into the types as well? Rather than `AddressNotRecognised`, should we have something like:

```
data AddressIsOurs = AddressIsOurs
                                    | AddressAmbiguousOwnership
```

probably not the best name in town, but would reflect the fact that the ownership is ambiguous: we really don't know if it's ours or not. `AddressNotRecognised` suggest to me something a bit _too_ strong.